packages/flutter_goldens_client/lib/skia_client.dart

```diff
@@ -94,31 +98,53 @@ class SkiaGoldClient {
   /// Prepares the local work space for golden file testing and calls the
   /// goldctl `auth` command.
   ///
-  /// This ensures that the goldctl tool is authorized and ready for testing. It
-  /// will only be called once for each instance of
-  /// [FlutterSkiaGoldFileComparator].
+  /// This ensures that the goldctl tool is authorized and ready for testing.
+  /// Used byt the [FlutterPostSubmitFileComparator] and the
+  /// [_AuthorizedFlutterPreSubmitComparator].
+  ///
+  /// Based on the current environment, the goldctl tool may be authorized by
+  /// with a service account for Cirrus testing, or through the context provided
+  /// by a luci environment.
   Future<void> auth() async {
     if (_clientIsAuthorized())
       return;
 
-    if (_serviceAccount.isEmpty) {
-      final StringBuffer buf = StringBuffer()
-        ..writeln('The Gold service account is unavailable.')
-        ..writeln('Without a service account, Gold can not be authorized.')
-        ..writeln('Please check your user permissions and current comparator.');
-      throw Exception(buf.toString());
-    }
-
-    final File authorization = workDirectory.childFile('serviceAccount.json');
-    await authorization.writeAsString(_serviceAccount);
+    List<String> authArguments;
+    String failureContext;
+
+    if (ci == 'luci') {
+      authArguments = <String>[
+        'auth',
+        '--work-dir', workDirectory
+          .childDirectory('temp')
+          .path,
+        '--luci',
+      ];
+      failureContext = 'Luci environments authenticate without a service '
```

Comment:
Technically, it uses the service account credentials exposed to it by Luci, but not via a file you specify, from a file in the LUCI_CONTEXT environment variable, but that's just a small detail that you don't have to worry about really.